File: jeongmin/스택_큐/기능개발.py
# progresses : 배포되어야 하는 순서대로의 작업
# speeds : 각 작업의 개발 속도
# 진도 100되면 배포 가능, 근데 앞의 것부터 배포된다.
# 각 배포당 몇 개의 기능이 배포되는가?

from collections import deque

def solution(progresses, speeds):
    progresses = deque(progresses)
    speeds = deque(speeds)
    answer = []
    while progresses:
        count = 0
        for i in range(len(progresses)):
            if progresses[i] < 100:
                if progresses[i] + speeds[i] >= 100:
                    progresses[i] = 100
                else:
                    progresses[i] += speeds[i]
        if progresses[0] == 100:
            while progresses and progresses[0] == 100:
                progresses.popleft()
                speeds.popleft()
                count += 1
                if len(progresses) == 0 or progresses[0] != 100:
                    answer.append(count)
    return answer

File: inho/프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        
        //큐야야약
        //큐로 다 넣는다.
        //수정해야되니까 List로 한다.
        int pSize = progresses.length;
        int []countList = new int[pSize];
        ArrayList<Integer> answerList = new ArrayList<>();
        for(int i=0; i<pSize; i++){
            int count = 0;
            while(progresses[i]  < 100){
                progresses[i] += speeds[i];
                count++;
            }
            countList[i] = count;
        }
        //answerList 에 갈겨 넣기 
        for(int i=0; i<pSize; i++){
            int returnTotalCount = 0;
            int movePoint = 0; 
            for(int j = i; j<pSize; j++){
                if(countList[i]>=countList[j]){
                    returnTotalCount ++;
                    movePoint = j;
                }else{
                    break;
                }
            }
            i = movePoint;
            answerList.add(returnTotalCount);
        }
        // for(Integer token : answerList){
        //     System.out.print(token+" ");
        // }
        int answerArrSize = answerList.size();
        int[] answer = new int[answerArrSize];
        for(int i=0; i<answerArrSize; i++){
            answer[i] = answerList.get(i);
        }
        return answer;
    }
}